clang/test/Driver/sycl-lto.cpp

```diff
@@ -8,7 +8,12 @@
 // RUN: not %clangxx -fsycl --offload-new-driver -foffload-lto=thin -fsycl-device-code-split=off %s -### 2>&1 | FileCheck -check-prefix=CHECK_SPLIT_ERROR %s
 // CHECK_SPLIT_ERROR: '-fsycl-device-code-split=off' is not supported when '-foffload-lto=thin' is set with '-fsycl'
 
-// Verify there's no error and we see the expected cc1 flags with the new offload driver.
+// Verify there's no error and we see the expected cc1 flags and tool invocations with the new offload driver.
 // RUN: %clangxx -fsycl --offload-new-driver -foffload-lto=thin %s -### 2>&1 | FileCheck -check-prefix=CHECK_SUPPORTED %s
 // CHECK_SUPPORTED: clang{{.*}} "-cc1" "-triple" "spir64-unknown-unknown" {{.*}} "-flto=thin" "-flto-unit"
```

Comment:
what will happen if users tries to use multiple targets here? Thanks

Reply:
i just tried it and it seems to create the jobs i expect, but we get a later error in `clang-linker-wrapper` because i haven't implemented anything there yet

```
[nsarnie@system llvm_earlysplit]$ clang++ -fsycl ../thinltotest/split.cpp --offload-new-driver -foffload-lto=thin -ccc-print-phases -fsycl-targets=spir64,spir64_gen
                  +- 0: input, "../thinltotest/split.cpp", c++, (host-sycl)
               +- 1: append-footer, {0}, c++, (host-sycl)
            +- 2: preprocessor, {1}, c++-cpp-output, (host-sycl)
         +- 3: compiler, {2}, ir, (host-sycl)
         |                       +- 4: input, "../thinltotest/split.cpp", c++, (device-sycl)
         |                    +- 5: preprocessor, {4}, c++-cpp-output, (device-sycl)
         |                 +- 6: compiler, {5}, ir, (device-sycl)
         |              +- 7: sycl-post-link, {6}, tempfiletable, (device-sycl)
         |           +- 8: file-table-tform, {7}, tempfilelist, (device-sycl)
         |           |- 9: backend, {8}, lto-bc, (device-sycl)
         |        +- 10: foreach, {8, 9}, lto-bc, (device-sycl)
         |     +- 11: file-table-tform, {10}, tempfilelist, (device-sycl)
         |  +- 12: offload, "device-sycl (spir64_gen-unknown-unknown)" {11}, tempfilelist
         |  |                    +- 13: input, "../thinltotest/split.cpp", c++, (device-sycl)
         |  |                 +- 14: preprocessor, {13}, c++-cpp-output, (device-sycl)
         |  |              +- 15: compiler, {14}, ir, (device-sycl)
         |  |           +- 16: sycl-post-link, {15}, tempfiletable, (device-sycl)
         |  |        +- 17: file-table-tform, {16}, tempfilelist, (device-sycl)
         |  |        |- 18: backend, {17}, lto-bc, (device-sycl)
         |  |     +- 19: foreach, {17, 18}, lto-bc, (device-sycl)
         |  |  +- 20: file-table-tform, {19}, tempfilelist, (device-sycl)
         |  |- 21: offload, "device-sycl (spir64-unknown-unknown)" {20}, tempfilelist
         |- 22: clang-offload-packager, {12, 21}, image, (device-sycl)
      +- 23: offload, "host-sycl (x86_64-unknown-linux-gnu)" {3}, "device-sycl (x86_64-unknown-linux-gnu)" {22}, ir
   +- 24: backend, {23}, assembler, (host-sycl)
+- 25: assembler, {24}, object, (host-sycl)
26: clang-linker-wrapper, {25}, image, (host-sycl)

````

Reply:
Awesome. Also, I think we are also adding changes in sycl-post-link to handle multiple targets. So, this flow will change eventually. Thanks for the details. Looks good.